core/src/main/scala/org/apache/spark/status/LiveEntity.scala

```diff
@@ -538,6 +538,14 @@ private class LiveRDD(val info: RDDInfo) extends LiveEntity {
     distributions.get(exec.executorId)
   }
 
+  def getPartitions(): Map[String, LiveRDDPartition] = {
+    partitions.toMap
```

Comment:
This makes a copy, no? Is there a need to make that copy?

Reply:
I think this just creates an immutable map and inserts all elements into it and so no copy of elements.

Reply:
But it makes a copy of the map, and that seems unnecessary. Of course it does not make a copy of the elements - it doesn't even know how to do that.

Reply:
Do we prefer just to simply return the mutable HashMap?

Reply:
Sure, it's an internal API. Listener code needs to avoid doing unnecessary things like copying stuff to avoid issues with dropping events.